src/ray/raylet_client/raylet_client.cc

```diff
@@ -459,25 +460,13 @@ void raylet::RayletClient::ReleaseUnusedBundles(
       });
 }
 
-void raylet::RayletClient::PinObjectIDs(
-    const rpc::Address &caller_address,
-    const std::vector<ObjectID> &object_ids,
-    const rpc::ClientCallback<rpc::PinObjectIDsReply> &callback) {
-  rpc::PinObjectIDsRequest request;
-  request.mutable_owner_address()->CopyFrom(caller_address);
-  for (const ObjectID &object_id : object_ids) {
-    request.add_object_ids(object_id.Binary());
-  }
-  pins_in_flight_++;
-  auto rpc_callback = [this, callback = std::move(callback)](
-                          Status status, const rpc::PinObjectIDsReply &reply) {
-    pins_in_flight_--;
-    callback(status, reply);
-  };
-  grpc_client_->PinObjectIDs(request, rpc_callback);
+void RayletClient::PinObjectIDs(const rpc::Address &caller_address,
+                                std::vector<ObjectID> object_ids,
```

Comment:
Won't this copy the `object_ids`? I think the previous version was fine, unless you saw a noticeable memory difference from it. Alternatively, you can add the object_ids directly to a buffered protobuf.

Reply:
The callsite was creating temporary vectors for ObjectIDs, so it would just move them into the internal buffer instead of copying them. However since we are only calling this with singular ObjectID, changed the signature to `const ObjectID &`.